src/core/plugin/plugin.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * @fileoverview Adviser Plugin.
+ *
+ */
+
+'use strict';
+
+const debug = require('debug')('adviser:plugin');
+
+const PluginTemplate = require('../external/plugin');
+const PluginContext = require('./plugin-context');
+const PluginSummary = require('./plugin-summary');
+
+/**
+ * Wrap individual plugins in the core
+ *
+ * @class Plugin
+ */
+class Plugin {
+  /**
+   *Creates an instance of Plugin.
+   * @param {String} id
+   * @param {Object} settings
+   * @param {Class} Core
+   * @memberof Plugin
+   */
+  constructor(id, settings, Core) {
+    this.processedRules = [];
+    this.id = id.toLowerCase();
+    this.settings = settings;
+
+    if (Core.prototype instanceof PluginTemplate) {
```

Comment:
This part seems a little weird, why do we need to pass in a Class reference here?